defcon/html/console_html.lua

```diff
@@ -125,12 +174,28 @@ return [[
 			outline: none;
 		}
 	</style>
-</head>
-<body bgcolor="#2D2F31" style="padding: 20px" onload="document.getElementById('command').focus()">
+	</head>
+]]
+local content = [[<body bgcolor="#2D2F31" style="padding: 20px" onload="document.getElementById('command').focus()">
 	<div class="box">
 		<textarea onkeydown="handlekeydown(event)" id="log" readonly></textarea>
 		<input type="text" id="command" onkeydown="handlekeydown(event)" placeholder="&gt;"/>
-	</div>
-</body>
-</html>
-]]
+		%s
+	</div>]]
+		
+local footer = [[</body>
+</html>]]
+
+function M.html(commands)
+	local data_list = ""
+	if next(commands) ~= nil then
+		data_list = [[<datalist id="commands">]]
+		for name, _ in pairs(commands) do
+			data_list = data_list .. [[<option value="]] .. name .. [[" />]]
+		end
+		data_list = data_list .. [[</datalist>]]
+	end
+	return header .. script .. style .. string.format(content, data_list) .. footer
```

Comment:
I split up the html content into sub sections. Mainly because there's a lot of `%` which string.format doesn't like.